src/Workspaces/CSharp/Portable/Execution/CSharpOptionsSerializationService.cs

```diff
@@ -84,8 +84,12 @@ public override ParseOptions ReadParseOptionsFrom(ObjectReader reader, Cancellat
             var languageVersion = (LanguageVersion)reader.ReadInt32();
             var preprocessorSymbolNames = reader.ReadArray<string>();
 
-            var options = new CSharpParseOptions(languageVersion, documentationMode, kind, preprocessorSymbolNames);
-            return options.WithFeatures(features);
+            var options = new CSharpParseOptions(languageVersion, documentationMode, kind);
+
+            // use WithPreprocessorSymbols instead of constructor to bypass preprocessor validation.
+            // https://github.com/dotnet/roslyn/issues/15797
+            return options.WithPreprocessorSymbols(preprocessorSymbolNames)
```

Comment:
Does this mean we have invalid options being returned? Will something else blow up down the line if we keep hold of invalid preprocessor symbol names? Someone from @dotnet/roslyn-compiler should confirm if this is fine.